draw laser line

gustavoneves12 opened the pull request:
initial code to draw laser line

gustavoneves12 committed:
initial commit
create a function to draw lines
create a ruby to test the plugin

gustavoneves12 committed:
draw the laser scan information
convert laser to point cloud add draw the lines

gustavoneves12 committed:
draw laser line using the laser scan data

doudou commented on viz/LaserLine.cpp:
```diff
@@ -3,40 +3,102 @@
+    /**
+     * set line colors
+     */
+    colors->push_back(osg::Vec4(0.1, 0.9, 0.1, 1.0));
```
Make the color configurable. Please.

doudou commented on viz/LaserLine.cpp:
```diff
@@ -3,40 +3,102 @@
+    double maxRange = data.maxRange / 1000.0; //convert millimeters to meters
+
+    vertices->clear();
+    for (int i = 0; i < ranges.size()-1; i++) {
```
This will lead to very bad things if the ranges are empty.

gustavoneves12 commented:
PING

gustavoneves12 committed:
set color configurable
create show range mode
check if ranges is not empty and the number of ranges is equal or greater than two

doudou commented on viz/LaserLine.cpp:
```diff
@@ -3,40 +3,137 @@
+
+    vertices->clear();
+
+    if (showRange) {
```
Remove this, please. There is already a plugin to display the ranges for a laser scanner, really no need to duplicate the functionality (at the cost of making this plugin more complicated).

doudou replied:
I'm merging it anyways, but please keep it in mind.
